packages/terra-list/src/List.jsx

```diff
@@ -58,10 +58,17 @@ const List = ({
   if (role && role.length > 0 && role !== 'none') {
     attrSpread.role = role;
   }
+  const allItems = children ? [] : undefined;
 
+  React.Children.forEach(children, (child) => {
```

Comment:
We do not want to clone each child item here and it will degrade performance and cause the list items to always re-render on update.

Reply:
Added condition in c967e3a

Reply:
Added Prop in df874d5